tests/mdev_configuration_allocation_test.go

```diff
@@ -237,7 +243,7 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 			}
 			tests.UpdateKubeVirtConfigValueAndWait(config)
 			By("Verify that the default mdev configuration didn't change")
-			checkAllMDEVCreated(desiredMdevTypeName, expectedInstancesNum)
+			Eventually(checkAllMDEVCreated(desiredMdevTypeName, expectedInstancesNum), 3*time.Minute, 15*time.Second).Should(BeInPhase(k8sv1.PodSucceeded))
```

Comment:
Now that I think about it, shouldn't this be an `Expect` instead ? You're already polling (via `PollImmediate`) within the `checkAllMDEVCreated` function.

Is there a reason for this nested retry behavior ?

Reply:
No. The meaning would be the same as before without the `Eventually`
The `PollImmediate` is just used to get how the check is completed (success/failure). The `Eventually` gives us the possibility to perform the check multiple times. This is what we need as what we are checking might not be completed when we check it(in this case it will show us failed but might succeed later).

Reply:
Really ? AFAIU, the `PollImmediate` - located inside the `checkAllMDEVCreated` func - will run the `waitForPod` function until it returns true, an error, or times out.

`waitForPod` in turn is blocking until the pod reaches either the `k8sv1.PodFailed` or `k8sv1.PodSucceeded` phases.

This means - afaiu - that you can just `Expect` once - by the time the expectation is asserted, the pod has already reached the `k8sv1.PodFailed` or `k8sv1.PodSucceeded` phases, failed to access the phases, or timed-out.

Reply:
> Really ? AFAIU, the `PollImmediate` - located inside the `checkAllMDEVCreated` func - will run the `waitForPod` function until it returns true, an error, or times out.

Yes, this is what I refer to as a check. Check = (spawn a pod with some logic and wait for completion)


> 
> `waitForPod` in turn is blocking until the pod reaches either the `k8sv1.PodFailed` or `k8sv1.PodSucceeded` phases.
> 
> This means - afaiu - that you can just `Expect` once - by the time the expectation is asserted, the pod has already reached the `k8sv1.PodFailed` or `k8sv1.PodSucceeded` phases, failed to access the phases, or timed-out.

Can you elaborate?

Reply:
Your "check" already made sure the pod's phase is either `k8sv1.PodFailed or k8sv1.PodSucceeded`.

Thus, since the pod's phase is already either `k8sv1.PodFailed or k8sv1.PodSucceeded` (or it flat out failed - err/timeout), you can just check once if you're in the expected phase. 

AFAIU, this `Eventually` will execute just once - because you have already polled it until it reached completion; might as well be explicit - i.e. you can just use `Expect` instead of `Eventually`.

Reply:
A new pod will be created with the `Eventually`, do you agree?

Reply:
Yes. But shouldn't this work consistently for all pods ? 

I might be missing something, but I don't understand why you need to create a new pod then perform the blocking check.

Reply:
I now realize the script checking the presence of mediated devices is a simple one-shot process, that requires a new pod to be created until it works - or times out.

You can resolve this comment.